fastlane/fastlane
# [deliver] Check if each non-framed screenshot has a framed counterpart

### Checklist

### Motivation and Context

It solves an open issue where if there is a single framed screenshot, all non-framed screenshots are skipped.

Resolves #15366 

### Description

When framed screenshots are detected and the current file is not framed, check to see if it has a framed counterpart. If it does, skip this unframed version. Otherwise, add this unframed version.

I created a new directory with framed and unframed files and tested just this feature.

### Testing Steps

I made a folder with a few .txt files that did and did not have the "_framed". Then, I tested my if statement like so:
```
files = Dir.glob('/Users/jess/testruby/thefiles/*.txt')
extensions = '{png,jpg,jpeg,txt}' 
framed_screenshots_found = Dir.glob("/Users/jess/testruby/thefiles/*_framed.#{extensions}").count > 0

if framed_screenshots_found
  puts "Framed screenshots are detected! 🖼 Non-framed screenshot files may be skipped. 🏃"
end

files.each do |file_path|
  puts "working on: #{file_path}..."
  is_framed = file_path.downcase.include?("_framed.")

  if framed_screenshots_found && !is_framed
    # check to see if there is a framed version of this file
    fext = File.extname(file_path)
    puts file_path.gsub("#{fext}","_framed#{fext}")
    if Dir.glob(file_path.gsub("#{fext}", "_framed#{fext}")).any?
      puts "🏃 Skipping screenshot file: #{file_path} because this file has a framed version"
      next
    end
  end

  # still here then have the file_path
  puts "i am keeping #{file_path}"
end
```
**The result here shows that the statement works - it skips bob.txt because there is a bob_framed.txt, but it keeps test.txt because there is no test_framed.txt:**

```
jess:testruby/ $ ruby test_file.rb                                    [5:56:43]
Framed screenshots are detected! 🖼 Non-framed screenshot files may be skipped. 🏃
working on: /Users/jess/testruby/thefiles/bob.txt...
/Users/jess/testruby/thefiles/bob_framed.txt
🏃 Skipping screenshot file: /Users/jess/testruby/thefiles/bob.txt because this file has a framed version
working on: /Users/jess/testruby/thefiles/bob_framed.txt...
i am keeping /Users/jess/testruby/thefiles/bob_framed.txt
working on: /Users/jess/testruby/thefiles/test.txt...
/Users/jess/testruby/thefiles/test_framed.txt
i am keeping /Users/jess/testruby/thefiles/test.txt
```

Please note that for the test to work with minimal setup, its code is slightly different than the version in the fastlane deliver file, which includes the watch and bold:
```
if framed_screenshots_found && !is_framed && !is_watch
            # check to see if there is a framed version of this file
            fext = File.extname(file_path)
            if Dir.glob(file_path.gsub("#{fext}", "_framed#{fext}")).any?
              UI.important("🏃 Skipping screenshot file: #{file_path} because this file has a framed version")
              next
            end
          end
```
